ddtrace/mocktracer/mockspan.go

```diff
@@ -7,282 +7,122 @@ package mocktracer // import "github.com/DataDog/dd-trace-go/v2/ddtrace/mocktrac
 
 import (
 	"fmt"
-	"sync"
 	"time"
 
 	"github.com/DataDog/dd-trace-go/v2/ddtrace"
 	"github.com/DataDog/dd-trace-go/v2/ddtrace/ext"
 	"github.com/DataDog/dd-trace-go/v2/ddtrace/tracer"
 )
 
-var _ ddtrace.Span = (*mockspan)(nil)
-var _ Span = (*mockspan)(nil)
-
-// Span is an interface that allows querying a span returned by the mock tracer.
-type Span interface {
-	// SpanID returns the span's ID.
-	SpanID() uint64
-
-	// TraceID returns the span's trace ID.
-	TraceID() uint64
-
-	// ParentID returns the span's parent ID.
-	ParentID() uint64
-
-	// StartTime returns the time when the span has started.
-	StartTime() time.Time
-
-	// FinishTime returns the time when the span has finished.
-	FinishTime() time.Time
-
-	// OperationName returns the operation name held by this span.
-	OperationName() string
-
-	// Tag returns the value of the tag at key k.
-	Tag(k string) interface{}
-
-	// Tags returns a copy of all the tags in this span.
-	Tags() map[string]interface{}
-
-	// Context returns the span's SpanContext.
-	Context() ddtrace.SpanContext
-
-	// StartChild starts a new child span with the given operation name and options.
-	StartChild(operationName string, opts ...ddtrace.StartSpanOption) ddtrace.Span
-
-	// Stringer allows pretty-printing the span's fields for debugging.
-	fmt.Stringer
+func newSpan(operationName string, cfg *ddtrace.StartSpanConfig) *tracer.Span {
+	return tracer.SpanStart(operationName, func(c *ddtrace.StartSpanConfig) {
+		*c = *cfg
+	})
 }
 
-func newSpan(t *mocktracer, operationName string, cfg *ddtrace.StartSpanConfig) *mockspan {
-	if cfg.Tags == nil {
-		cfg.Tags = make(map[string]interface{})
-	}
-	if cfg.Tags[ext.ResourceName] == nil {
-		cfg.Tags[ext.ResourceName] = operationName
-	}
-	s := &mockspan{
-		name:   operationName,
-		tracer: t,
-	}
-	if cfg.StartTime.IsZero() {
-		s.startTime = time.Now()
-	} else {
-		s.startTime = cfg.StartTime
-	}
-	id := cfg.SpanID
-	if id == 0 {
-		id = nextID()
-	}
-	s.context = &spanContext{spanID: id, traceID: id, span: s}
-	if ctx, ok := cfg.Parent.(*spanContext); ok {
-		if ctx.span != nil && s.tags[ext.ServiceName] == nil {
-			// if we have a local parent and no service, inherit the parent's
-			s.SetTag(ext.ServiceName, ctx.span.Tag(ext.ServiceName))
-		}
-		if ctx.hasSamplingPriority() {
-			s.SetTag(ext.SamplingPriority, ctx.samplingPriority())
-		}
-		s.parentID = ctx.spanID
-		s.context.priority = ctx.samplingPriority()
-		s.context.hasPriority = ctx.hasSamplingPriority()
-		s.context.traceID = ctx.traceID
-		s.context.baggage = make(map[string]string, len(ctx.baggage))
-		ctx.ForeachBaggageItem(func(k, v string) bool {
-			s.context.baggage[k] = v
-			return true
-		})
-	}
-	for k, v := range cfg.Tags {
-		s.SetTag(k, v)
-	}
-	return s
+type Span struct {
+	*tracer.Span
 }
```

Comment:
Is this really necessary? This struct has a single pointer embed... of a type by the same leaf name.

Or is this supposed to be the new `MockSpan` type? In which case perhaps this needs renaming?

Reply:
Yes, this is the new `MockSpan`. I can change it back to `MockSpan` if you prefer, but I was hoping to eliminate the stuttering `mocktracer.MockSpan`.